sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2156,10 +2156,12 @@ def task_worker(self, stopping_event, sfp_error_event):
 
                                 if rc != SFP_EEPROM_NOT_READY:
                                     post_port_dom_threshold_info_to_db(logical_port, self.port_mapping, self.xcvr_table_helper.get_dom_threshold_tbl(asic_index))
-                                    media_settings_parser.notify_media_setting(logical_port, transceiver_dict, self.xcvr_table_helper.get_app_port_tbl(asic_index), self.xcvr_table_helper.get_cfg_port_tbl(asic_index), self.port_mapping)
+                                    media_settings_parser.notify_media_setting(logical_port, transceiver_dict, self.xcvr_table_helper, self.port_mapping)
                                     transceiver_dict.clear()
                             elif value == sfp_status_helper.SFP_STATUS_REMOVED:
                                 helper_logger.log_notice("{}: Got SFP removed event".format(logical_port))
+                                state_port_table = self.xcvr_table_helper.get_state_port_tbl(asic_index)
```

Comment:
@mihirpat1 is there any specific reason to not delete this key on optics removal?

Reply:
@prgeor I am not deleting the key on optics removal since as part of xcvrd boot-up, we are setting the key to `NPU_SI_SETTINGS_DEFAULT` for all ports (irrespective of transceiver presence). Hence, in order to keep the state consistent across device boot-up and optics OIR, I am setting the value as opposed to deleting the key.
Also, differentiating if a port requires NPU SI settings as part of device boot-up will put overhead on xcvrd main thread (via `initialize_port_init_control_fields_in_port_table`) to read the EEPROM and conclude if the optics requires NPU SI settings and set the value accordingly. Hence, I am setting the value for this key to `NPU_SI_SETTINGS_DEFAULT` for all ports and have SfpStateUpdateTask (and CmisManagerTask in future) to conclude if the optics requires NPU SI settings.